### PR TITLE
api: add nvim_get_helptags

### PR DESCRIPTION
**Pull request by vigoux**
After looking at some really hacky code @tjdevries pointed me to, I thought that we really should have this.
This allows to search helptags easily.
Hopefully this is not too messy.

### CODE REVIEW

File: src/nvim/api/vim.c [hunk 1]
```diff
@@ -2767,3 +2768,29 @@ void nvim__set_luahl(DictionaryOf(LuaRef) opts, Error *err)
 error:
   clear_luahl(true);
 }
+
+/// Search the helptags for a pattern
+///
+/// @param pattern the pattern to search
+/// @param[out] err error if any
+Array nvim_get_helptags(String pattern, Error *err)
```
Comment (bfredl) [inline src/nvim/api/vim.c:2776]: Do we want this to be `nvim_get_tags` already? Supporting only helptags is OK for now, but should it be spelled as `nvim_get_tags("pattern", {source="help"})` or something like that, so it is easier to extend later?

Comment (vigoux) [reply src/nvim/api/vim.c:2776]: That sounds like a good idea.
The thing that needs to be adressed too is the return value of the function.
After some discussion with @tjdevries we thought that a `taglist()`-like return value can be great, looking like this :
```
[
  {
    tag : String,
    fname : String,
    lnum : Integer
  }
]
```

**Comment by justinmk** [comment]
Shoudl this be part of `nvim_get_api` ? https://github.com/neovim/neovim/pull/12040

**Comment by vigoux** [comment]
I think that this should be part of `nvim_get_tags` so that we mimic how tags work in nvim (the viml part of it)

**Comment by justinmk** [comment]
For reference, how does this compare to `taglist()` ? Do we want a `opt` param  and maybe return a `Dict` for potential future expansion?

**Comment by vigoux** [comment]
I think that this can be a reimplementaion of taglist exposed to the API for now, and maybe something more configurable later

**Comment by justinmk** [comment]
> I think that this can be a reimplementaion of taglist exposed to the API for now, and maybe something more configurable later

Needs a stronger motivation. 

Example of using taglist(): https://github.com/neovim/neovim/blob/1ee905a63aa9f3d6873ec75a0954866d8660699e/scripts/gen_help_html.lua#L625-L638

One thing that's weird about taglist() is that you need to visit a buffer before using it. But this PR doesn't look like it changes anything there.

**Pull request closed without merging**